src/components/ScholarshipList.jsx

```diff
@@ -39,6 +40,15 @@ const useStyles = makeStyles((theme) => ({
 
 function ScholarshipList({ scholarships }) {
   const classes = useStyles();
+  const [shareDialogOpen, setShareDialogOpen] = useState(false);
+  const showShareDialog = () => setShareDialogOpen(true);
```

Comment:
This function isn't used as a variable anywhere. Nor is it invoked multiple times. You may as well inline the logic the only place it's inovked.

Reply:
Not sure what you mean by that? Place lines 43-50 in the shareFn function?

Reply:
What's the point of defining some function
```js
const someFunction = () => setSomeState(foo);
```
and then invoking it once:
```js
someFunction();
```
when you can just inline the logic:
```js
setSomeState(foo);
```
What's the purpose of `someFunction`? It's never used as a variable anywhere nor is it ever invoked multiple times.

Reply:
Ok I see what you were referring to, changes are made